core/src/main/scala/kafka/utils/Exit.scala

```diff
@@ -34,20 +34,34 @@ object Exit {
     throw new AssertionError("halt should not return, but it did.")
   }
 
+  def addShutdownHook(name: String, shutdownHook: => Unit): Unit = {
+    JExit.addShutdownHook(name, () => shutdownHook)
+  }
+
   def setExitProcedure(exitProcedure: (Int, Option[String]) => Nothing): Unit =
     JExit.setExitProcedure(functionToProcedure(exitProcedure))
 
   def setHaltProcedure(haltProcedure: (Int, Option[String]) => Nothing): Unit =
     JExit.setHaltProcedure(functionToProcedure(haltProcedure))
 
+  def setShutdownHookAdder(shutdownHookAdder: (String, => Unit) => Unit): Unit = {
+    JExit.setShutdownHookAdder(functionToShutdownHookAdder(shutdownHookAdder))
```

Comment:
Couldn't we do something like:
```
JExit.setShutdownHookAdder((name, runnable) => shutdownHookAdder(name, runnable.run()))
```
instead of a separate  `functionToShutdownHookAdder` method?